### PR TITLE
Enhancements in clarity to access-application-cluster/connecting-frontend-backend.md

### LINKED ISSUE #25818: Improvement for k8s.io/docs/tasks/access-application-cluster/connecting-frontend-backend/
Replace 'hello' with something else so it's easier to distinguish where which name is being used.. hard to follow the guide.
An example I can give is from the 'Creating the frontend' section, in particular the nginx configuration.

```
upstream hello {
    server hello;
}

server {
    listen 80;

    location / {
        proxy_pass http://hello;
    }
}
```
Since both the backend and frontend are named hello, it's unclear which 'hello' this refers to.

### PR DESCRIPTION
**Pull request by murillodigital**
fixes #25818

### CODE REVIEW

**Comment by murillodigital** [comment]
@Aut0R3V @sftim I've made changes to the overall document towards more accurate description and better clarity, including renaming the manifests and config files to better match their place inside the tutorial. I do have one question, the nginx config in the docs is already baked inside an already published dockerimage `gcr.io/google-samples/hello-frontend:1.0` the change to service name and related change to match in the nginx config would effectively be a breaking change for anybody applying the new manifests using the existing dockerimage. How do we go about publishing a new version to match the documentation?

File: content/en/docs/tasks/access-application-cluster/connecting-frontend-backend.md [hunk 1]
```diff
@@ -87,50 +89,59 @@ Events:
...
+Create the `backend` Service:
 
 ```shell
-kubectl apply -f https://k8s.io/examples/service/access/hello-service.yaml
+kubectl apply -f https://k8s.io/examples/service/access/backend-service.yaml
 ```
 
-At this point, you have a backend Deployment running, and you have a
-Service that can route traffic to it.
+At this point, you have a `backend` Deployment running seven replicas of your `hello`
```
Comment (tengqm) [inline content/en/docs/tasks/access-application-cluster/connecting-frontend-backend.md:111]: Sounds to me that 7 replicas is an overkill. Can/shall we reduce it to 3?

File: content/en/docs/tasks/access-application-cluster/connecting-frontend-backend.md [hunk 2]
```diff
@@ -87,50 +89,59 @@ Events:
...
+application, and you have a Service that can route traffic to them. However, this 
+service is neither available nor resolvable outside the cluster yet.
 
 ## Creating the frontend
 
-Now that you have your backend, you can create a frontend that connects to the backend.
-The frontend connects to the backend worker Pods by using the DNS name
-given to the backend Service. The DNS name is "hello", which is the value
-of the `name` field in the preceding Service configuration file.
+Now that you have your backend running, you can create a frontend that is available
```
Comment (tengqm) [inline content/en/docs/tasks/access-application-cluster/connecting-frontend-backend.md:117]: "available" may be a little ambiguous?

```suggestion
Now that you have your backend running, you can create a frontend that is accessible from 
```

File: content/en/docs/tasks/access-application-cluster/connecting-frontend-backend.md [hunk 3]
```diff
@@ -87,50 +89,59 @@ Events:
...
-Similar to the backend, the frontend has a Deployment and a Service. The
-configuration for the Service has `type: LoadBalancer`, which means that
-the Service uses the default load balancer of your cloud provider.
+{{< codenew file="service/access/frontend-nginx.conf" >}}
 
-{{< codenew file="service/access/frontend.yaml" >}}
+Similar to the backend, the frontend has a Deployment and a Service. An important
+difference to notice between the backend and frontend services, is that the
+configuration for the frontend Service has `type: LoadBalancer`, which means that
+the Service uses the default load balancer of your cloud provider and will be
```
Comment (tengqm) [inline content/en/docs/tasks/access-application-cluster/connecting-frontend-backend.md:133]:
```suggestion
the Service uses a load balancer of your cloud provider and will be
```

**Comment by tengqm** [comment]
> the change to service name and related change to match in the nginx config would effectively be a breaking change for anybody applying the new manifests using the existing dockerimage.

Will the new "nginx.conf" override the defaults in the docker image? If not, this problem may become a blocker.

**Comment by murillodigital** [comment]
> 
> 
> > the change to service name and related change to match in the nginx config would effectively be a breaking change for anybody applying the new manifests using the existing dockerimage.
> 
> Will the new "nginx.conf" override the defaults in the docker image? If not, this problem may become a blocker.

The Dockerfile does indeed replace the default conf file with our frontend-nginx.conf:

```
RUN rm /etc/nginx/conf.d/default.conf
COPY frontend-nginx.conf /etc/nginx/conf.d
```

 The problem is, the current image will attempt to proxy traffic to a hostname named `hello`, which with the introduced changes no longer exists (the new backend service name is `backend`)

File: content/en/docs/tasks/access-application-cluster/connecting-frontend-backend.md (hunk 1 above)
Comment (murillodigital) [reply content/en/docs/tasks/access-application-cluster/connecting-frontend-backend.md]: Agreed, reduced to 3 (here as well as deployment manifest and sample apply output)

**Comment by tengqm** [comment]
> The problem is, the current image will attempt to proxy traffic to a hostname named hello, which with the introduced changes no longer exists (the new backend service name is backend)

In that case, we may need to change the backend service name back to `"hello"`?

**Comment by murillodigital** [comment]
> 
> 
> > The problem is, the current image will attempt to proxy traffic to a hostname named hello, which with the introduced changes no longer exists (the new backend service name is backend)
> 
> In that case, we may need to change the backend service name back to `"hello"`?

Either that or replace the current image or update one with a newer version, which is what I would prefer.

I don't think the name `hello` is ideal to properly communicate what the hostname is pointing at, nor consistent with the rest of the nomenclature (`backend` vs `frontend` vs `app=hello`).

My preferred approach is build/push a new image `gcr.io/google-samples/hello-frontend:1.1`  to match the changes in this PR. and update the deployment manifest accordingly - for that, I would need somebody with push access to the google-samples gcr repository.

**Comment by tengqm** [comment]
> My preferred approach is build/push a new image gcr.io/google-samples/hello-frontend:1.1 to match the changes in this PR. 

Doable, but we need to find the person for the image upgrade and we will need to hold this PR till the new image is published.

**Comment by murillodigital** [comment]
> 
> 
> > My preferred approach is build/push a new image gcr.io/google-samples/hello-frontend:1.1 to match the changes in this PR.
> 
> Doable, but we need to find the person for the image upgrade and we will need to hold this PR till the new image is published.

Patience is a virtue 🙂. How would I escalate/get a hold of somebody with access and cycles to support? Is there something other than delay that I may be overlooking?

**Comment by tengqm** [comment]
Just tried to google the source for this image but failed. From gcr, I can see the image was created in 2016 and never updated since its initial version. Anyway, I'm okay with upgrading the image too, so long the whole example works.

**Comment by tengqm** [comment]
Is this the source code? https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/tree/master/hello-app

**Comment by murillodigital** [comment]
> 
> 
> Is this the source code? https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/tree/master/hello-app

You don't need to change the actual hello-app, only the nginx frontend. The Dockerfile to build it is in the docs repo itself, it's nothing but an nginx image with a swapped config file.

The Dockerfile is at `/content/en/examples/service/access` - you'd just have to build it from that path with a tag `gcr.io/google-samples/hello-frontend:1.1` and push it.

**Comment by tengqm** [comment]
@murillodigital I see. Thanks.

**Comment by murillodigital** [comment]
> 
> 
> @murillodigital I see. Thanks.

Hi there @tengqm ! Just doing a quick drive by to see if you had been able to make some progress on the image building? Let me know if there's anything I can do to help out. Hope you have a phenomenal weekend!

**Comment by tengqm** [comment]
@murillodigital Unfortunately, I don't have access to the image repo.

**Comment by murillodigital** [comment]
> 
> 
> @murillodigital Unfortunately, I don't have access to the image repo.

Not a problem, do you think it's reasonably possible to find somebody with access, or would you advise that, although less than ideal from my perspective, we simply revert the service name (all other changes would still apply)?

**Comment by tengqm** [comment]
@murillodigital I don't know who has this access. Maybe keep the service name as `hello` is a simpler alternative.

**Comment by murillodigital** [comment]
> 
> 
> @murillodigital I don't know who has this access. Maybe keep the service name as `hello` is a simpler alternative.

Cool, just renamed the backend service back to `hello` and updated references for consistency, fixed some wording around the term looking to keep clarity - there are no blockers to merging anymore.

**Comment by murillodigital** [comment]
@tengqm  - quick ping on this, this should be good to go with my last change.
